test/per_file_coverage.sh

```diff
@@ -21,7 +21,7 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/tcp:94.6"
 "source/common/thread:0.0" # Death tests don't report LCOV
 "source/common/watchdog:58.6" # Death tests don't report LCOV
-"source/exe:91.5"
+"source/exe:78.5" # main function is not covered
```

Comment:
Hm. What changed as part of this PR which caused source/exe coverage to drop?

Reply:
There's a single envoy_sh_test that was covering main(). It requires a full envoy binary which is over 2GB linker limit with coverage at this point.

Reply:
ok so let's add a smaller test which covers main, rather than dropping coverage :-)

Reply:
The problem is not coming up with a test, but with linking a minimal Envoy binary. With new protobuf and coverage enabled it fails to link due to static data being over 4Gb.

Reply:
right, if we sharply limit extensions is it still a problem?

Reply:
to be a bit more clear, as the person who has been tearing out all possible extensions to reduce E-M binary size =P
I suspect whatever tests because of the binary size increase, can be addressed by switching the target from envoy_main_common_with_core_extensions_lib to :main_common_lib and explicitly adding the extensions needed for that test.  There's a lot from the validation server to health checkers we probably don't need for a basic main common test

Reply:
Yeah @alyssawilk that works, see my PR https://github.com/envoyproxy/envoy/pull/31456

Reply:
sweet!  dismissed the comment and removed myself as reviewer since I was only there as coverage shepherd. :-)

Reply:
That being said, it looks like this PR is still lowering the limit in per_file_coverage.sh. Let's make sure we get that changed before we land?

Reply:
Even with the minimum number of extensions the `test/integration:run_envoy_test` does not link in coverage build.

Reply:
Hm. This is quite surprising. I would have expected `:main_common_lib` to remove a LARGE number of extensions and I wouldn't have expected an update to protobuf to add *That* much new code. Is there a patch set I could look at which uses `:main_common_lib` but does not link?

Reply:
yeah especially given @kyessenov said it was fixed?

Reply:
It was not fixed. I have tried with the new library and the link is still failing.